sdk/src/common/core.cc

```diff
@@ -0,0 +1,12 @@
+#include "opentelemetry/nostd/variant.h"
```

Comment:
`core.cc` contains the only method `absl::variant_internal::ThrowBadVariantAccess` that would otherwise be necessary for inclusion via `${CORE_RUNTIME_LIBS}` which is defined as `set(CORE_RUNTIME_LIBS absl::any absl::base absl::bits absl::city)` ... We could avoid this by borrowing `absl::variant` implementation and plugging it in instead of MPark.Variant. That is the reason why I moved the MPark.Variant out into separate folder, to replace it with variant from Abseil. This will be needed to solve these issues: #215 #314

Reply:
Since `std::variant` is available since  C++17, we only need non-standard variant for older compilers, such as Visual Studio 2015 (where MPark variant does not compile), OR for older gcc-4.8 (Ubuntu-14.04).. Whereas any other more recent LTS version of Linux, e.g. Ubuntu-16.04 comes with gcc-5, where we won't be needing the Abseil at all.